packages/slate-react/src/components/leaf.js

```diff
@@ -125,7 +125,7 @@ class Leaf extends React.Component {
     // COMPAT: If the text is empty and it's the only child, we need to render a
     // line break to get the block to have the proper height.
     if (text == '' && parent.kind == 'block' && parent.text == '') {
-      return <span data-slate-empty-block>{'\n'}</span>
+      return <span data-slate-empty-block>{'\u200B'}</span>
```

Comment:
This works! For the remaining two failing test case, just use

```
<span data-slate-empty-block="true"></span>
```

instead of

```
<span data-slate-empty-block="true">\n</span>
```

and I believe that'll be adequate.

Reply:
Good catch! Forgot to run the tests 😄